src/test/java/org/opensearch/plugin/insights/core/service/categorizor/QueryShapeGeneratorTests.java

```diff
@@ -0,0 +1,106 @@
+
+        String shapeShowFieldsFalse = QueryShapeGenerator.buildShape(sourceBuilder, false);
+        String expectedShowFieldsFalse = "bool\n"
+            + "  must:\n"
+            + "    term\n"
+            + "  filter:\n"
+            + "    match\n"
+            + "    range\n"
+            + "  should:\n"
+            + "    regexp\n";
+        assertEquals(expectedShowFieldsFalse, shapeShowFieldsFalse);
+    }
+
+    public void testAggregationShape() {
+        SearchSourceBuilder sourceBuilder = new SearchSourceBuilder();
+        sourceBuilder.aggregation(new TermsAggregationBuilder("agg1").userValueTypeHint(ValueType.STRING).field("type"));
+        sourceBuilder.aggregation(new TermsAggregationBuilder("agg2").userValueTypeHint(ValueType.STRING).field("model"));
+        sourceBuilder.aggregation(
+            new TermsAggregationBuilder("agg3").userValueTypeHint(ValueType.STRING)
+                .field("key")
+                .subAggregation(new TermsAggregationBuilder("child-agg1").userValueTypeHint(ValueType.STRING).field("key.sub1"))
+                .subAggregation(new TermsAggregationBuilder("child-agg2").userValueTypeHint(ValueType.STRING).field("key.sub2"))
+        );
+
+        String shapeShowFieldsTrue = QueryShapeGenerator.buildShape(sourceBuilder, true);
+        String expectedShowFieldsTrue = "aggregation:\n"
+            + "  terms [key]\n"
+            + "    aggregation:\n"
+            + "      terms [key.sub1]\n"
+            + "      terms [key.sub2]\n"
+            + "  terms [model]\n"
+            + "  terms [type]\n";
+        assertEquals(expectedShowFieldsTrue, shapeShowFieldsTrue);
+
+        String shapeShowFieldsFalse = QueryShapeGenerator.buildShape(sourceBuilder, false);
+        String expectedShowFieldsFalse = "aggregation:\n"
+            + "  terms\n"
+            + "  terms\n"
+            + "  terms\n"
+            + "    aggregation:\n"
+            + "      terms\n"
+            + "      terms\n";
+        assertEquals(expectedShowFieldsFalse, shapeShowFieldsFalse);
+    }
+
+    public void testSortShape() {
+        SearchSourceBuilder sourceBuilder = new SearchSourceBuilder();
+        sourceBuilder.sort("color", SortOrder.DESC);
+        sourceBuilder.sort("vendor", SortOrder.DESC);
+        sourceBuilder.sort("price", SortOrder.ASC);
+        sourceBuilder.sort("album", SortOrder.ASC);
+
+        String shapeShowFieldsTrue = QueryShapeGenerator.buildShape(sourceBuilder, true);
+        String expectedShowFieldsTrue = "sort:\n" + "  asc [album]\n" + "  asc [price]\n" + "  desc [color]\n" + "  desc [vendor]\n";
+        assertEquals(expectedShowFieldsTrue, shapeShowFieldsTrue);
+
+        String shapeShowFieldsFalse = QueryShapeGenerator.buildShape(sourceBuilder, false);
+        String expectedShowFieldsFalse = "sort:\n" + "  asc\n" + "  asc\n" + "  desc\n" + "  desc\n";
+        assertEquals(expectedShowFieldsFalse, shapeShowFieldsFalse);
+    }
```

Comment:
Can add tests for:
1. querybuilders not present in the field map or field missing
2. tests containing pipeline aggregations
3. large complex query shapes with nested indentation
4. making sure the normalization is working properly in terms of sorting of field_names and query operations, etc

Reply:
1. `BoolQueryBuilder` -> `bool []` is present. Plus added a few other aggs
2. Updated existing agg test
3. Added testComplexSearch()
4. This is tested